packages/widget_toolkit_qr/lib/src/lib_qr_scanner/views/qr_scanner_component.dart

```diff
@@ -144,38 +160,25 @@ typedef AppErrorCallback = Widget? Function(
 
 class _QRBarScannerCamera extends StatelessWidget {
   const _QRBarScannerCamera({
-    required this.qrCodeCallback,
+    this.controller,
     required this.onError,
-    this.formats,
-    this.notStartedBuilder,
-    this.offscreenBuilder,
+    required this.onDetect,
   });
 
-  final WidgetBuilder? notStartedBuilder;
-  final WidgetBuilder? offscreenBuilder;
-  final ValueChanged<String?> qrCodeCallback;
   final AppErrorCallback onError;
-  final List<BarcodeFormats>? formats;
+  final MobileScannerController? controller;
+  final Function(BarcodeCapture)? onDetect;
 
   @override
-  Widget build(BuildContext context) => QRBarScannerCamera(
-        formats: formats,
-        onError: (ctx, error) => FutureBuilder(
-          future: _getError(ctx, error.toString()),
-          initialData: const SizedBox(),
-          builder: (BuildContext context, AsyncSnapshot<Widget> snapshot) =>
-              snapshot.data!,
-        ),
-        notStartedBuilder: notStartedBuilder,
-        offscreenBuilder: offscreenBuilder,
-        qrCodeCallback: (code) => qrCodeCallback(code),
+  Widget build(BuildContext context) => MobileScanner(
+        controller: controller,
+        onDetect: onDetect,
+        errorBuilder: (context, error, widget) => _getError(context, error),
```

Comment:
Do we need to handle [app lifecycle changes](https://pub.dev/packages/mobile_scanner#usage) or is this not necessary?

Reply:
It is not necessary to use these. In the examples that are provided by the `mobile_scanner` developer there are multiple ways to use the plugin

https://github.com/juliansteenbakker/mobile_scanner/blob/master/example/lib/mobile_scanner_overlay.dart